modules/add-contextual-data/tests/test_context_info_db.c

```diff
@@ -267,6 +289,34 @@ test_import_with_valid_csv(void)
 }
 
 static void
+test_import_with_valid_csv_crlf(void)
+{
+  gchar csv_content[] = "selector1,name1,value1\r\n"
+                        "selector1,name1.1,value1.1";
+
+  TestNVPair expected_nvpairs_selector1[2] =
+  {
+    {.name = "name1",.value = "value1"},
+    {.name = "name1.1",.value = "value1.1"},
+  };
+
+  _assert_import_csv_with_single_selector(csv_content, "selector1", expected_nvpairs_selector1, 2);
```

Comment:
I didn't specifically ask you to change it, because it kind of works correctly at present. In the null-terminated form, the null terminator could also be sometimes forgotten about. Although it would potentially be caught later on in case garbage resides after the array, there is no guarantee here.

I don't have a clear cut solution I could recommend, but you probably already know about clean magical macro workarounds like this:
https://github.com/balabit/syslog-ng/blob/c0bee2a8c5164f6968d2b6e2c3f4a4c0d8d9d0cd/modules/kvformat/tests/test_kv_scanner.c#L126

A much less hygienic workaround would be to introduce a macro that simply constructs `array, sizeof(array)/sizeof(array[0])`, and paste that instead of the argument.